packages/core/core/src/RequestTracker.js

```diff
@@ -675,7 +675,7 @@ export class RequestGraph extends ContentGraph<
           requestGraphEdgeTypes.invalidated_by_create_above,
         ) &&
         isDirectoryInside(
-          path.dirname(fromProjectPathRelative(matchNode.value.filePath)),
+          fromProjectPathRelative(matchNode.value.filePath),
```

Comment:
This change reduces the number of invalidations we need because `aboveFilePath` refers to a directory rather than a file.